definition_validator/validator.cc

```diff
@@ -134,6 +134,28 @@ void validateOverriding(const core::GlobalState &gs, core::SymbolRef method) {
     ENFORCE(klass.data(gs)->isClass());
     auto klassData = klass.data(gs);
     InlinedVector<core::SymbolRef, 4> overridenMethods;
+
+    if (klassData->isClassInterface() && method.data(gs)->isPrivate()) {
+        if (auto e = gs.beginError(method.data(gs)->loc(), core::errors::Resolver::NonPublicAbstract)) {
+            e.setHeader("Interface method `{}` cannot be private", method.show(gs));
```

Comment:
why not?

Reply:
I can imagine arguments both ways, but the real reason is because this was one of the checks that the runtime did before: https://github.com/sorbet/sorbet/blob/master/gems/sorbet-runtime/test/types/interface_validation.rb#L88-L123

I'd be fine with changing it—I know that Java originally disallowed private methods in interfaces and eventually allowed them, as well—but I was just adhering to the semantics of the existing checks.

Reply:
> I was just adhering to the semantics of the existing checks.

+1, I like that.
Let's add a comment saying that it's restricted to be conservative.

>I know that Java originally disallowed private methods in interfaces and eventually allowed them, as well

Note that in Java private methods are not inherited and do not use virtual dispatch. In ruby they are and they do use virtual dispatch. Reasoning from Java doesn't apply.